### PR TITLE
Feature/cipher suites

### LINKED ISSUE #244: Disable support for weak Ciphers using HTTPS
Our security team have flagged up webhook due to it's support for 3DES encryption. Is there anyway to disable support? 

Output of [sslscan](https://github.com/rbsec/sslscan) against https enabled webhook.
```
  Supported Server Cipher(s):
Preferred TLSv1.2  128 bits  ECDHE-RSA-AES128-GCM-SHA256
Accepted  TLSv1.2  256 bits  ECDHE-RSA-AES256-GCM-SHA384
Accepted  TLSv1.2  128 bits  ECDHE-RSA-AES128-SHA
Accepted  TLSv1.2  256 bits  ECDHE-RSA-AES256-SHA
Accepted  TLSv1.2  128 bits  AES128-GCM-SHA256
Accepted  TLSv1.2  256 bits  AES256-GCM-SHA384
Accepted  TLSv1.2  128 bits  AES128-SHA
Accepted  TLSv1.2  256 bits  AES256-SHA
Accepted  TLSv1.2  112 bits  ECDHE-RSA-DES-CBC3-SHA
Accepted  TLSv1.2  112 bits  DES-CBC3-SHA
Preferred TLSv1.1  128 bits  ECDHE-RSA-AES128-SHA
Accepted  TLSv1.1  256 bits  ECDHE-RSA-AES256-SHA
Accepted  TLSv1.1  128 bits  AES128-SHA
Accepted  TLSv1.1  256 bits  AES256-SHA
Accepted  TLSv1.1  112 bits  ECDHE-RSA-DES-CBC3-SHA
Accepted  TLSv1.1  112 bits  DES-CBC3-SHA
Preferred TLSv1.0  128 bits  ECDHE-RSA-AES128-SHA
Accepted  TLSv1.0  256 bits  ECDHE-RSA-AES256-SHA
Accepted  TLSv1.0  128 bits  AES128-SHA
Accepted  TLSv1.0  256 bits  AES256-SHA
Accepted  TLSv1.0  112 bits  ECDHE-RSA-DES-CBC3-SHA
Accepted  TLSv1.0  112 bits  DES-CBC3-SHA
```

### PR DESCRIPTION
**Pull request by moorereason**
Default to TLS 1.2 and secure cipher suites.

Built for Go 1.13. Code in cipher_suites.go taken from Go tip commit
0ee22d9, which is scheduled for the upcoming Go 1.14 release.  Once Go
1.14 is released, we can remove this file and use the stdlib.

Fixes #244

### CODE REVIEW

File: webhook.go [hunk 1]
```diff
@@ -39,6 +40,8 @@ var (
 	cert               = flag.String("cert", "cert.pem", "path to the HTTPS certificate pem file")
 	key                = flag.String("key", "key.pem", "path to the HTTPS certificate private key pem file")
 	justDisplayVersion = flag.Bool("version", false, "display webhook version and quit")
+	tlsMinVersion      = flag.String("tls-min-version", "1.2", "minimum TLS version (1.0, 1.1, 1.2, 1.3)")
```
Comment (adnanh) [inline webhook.go]: Can you add these flags to [docs/Webhook-Parameters.md](https://github.com/adnanh/webhook/blob/master/docs/Webhook-Parameters.md)?

Comment (moorereason) [reply webhook.go]: Done.

**Comment by moorereason** [comment]
Regarding my comment about removing the `cipher_suites.go` file after Go 1.14, we would need to break out some of this code into build tag protected files to continue supporting Go 1.12 & 1.13, but we can deal with that later.

**Comment by adnanh** [comment]
Should we add the fact that we need at least 1.12 Go version in order to run this project to the README.md?

**Pull request merged**